src/ints/bios.cpp

```diff
@@ -1269,7 +1269,10 @@ class BIOS final : public Module_base{
 		for (const auto c : "01/01/92")
 			phys_writeb(0xffff5 + i++, static_cast<uint8_t>(c));
 
-		phys_writeb(0xfffff, 0x55); // signature
+		if (machine==MCH_TANDY)
+			phys_writeb(0xfffff, 0xff); // Needed for Ninja (1986)
+		else
+			phys_writeb(0xfffff, 0x55); // signature
```

Comment:
After attributing the first commit to NRS, I suggest a second commit bring it in line with Staging's approach, specifically to de-magic variables.

So this second commit will be your own.

The `0xfffff` address to a self-documenting `constexpr uint32_t`, say `bios_machine_signature_location`.

The `0xff` is the signature for Tandy PCs and `0x55` is for IBM/XT/AT PCs can also be assigned into self-documenting variable, perhaps: `const uint8_t machine_signature`

```c++
constexpr uint32_t machine_signature_location = 0xfffff;
const auto uint8_t machine_signature = (machine == MCH_TANDY) ? 0xff : 0x55;
phys_writeb(machine_signature_location, machine_signature);
```

Feel free to adjust with better names :+1: